File: tree/reverse/invert_tree.go
package reverse

import "trouble_solver/tree"

func invertTree(root *tree.TreeNode) *tree.TreeNode {
	if root == nil {
		return nil
	}
	if root.Left == nil && root.Right == nil {
		return root
	}
	leftTree := invertTree(root.Left)
	rightTree := invertTree(root.Right)
	root.Left, root.Right = rightTree, leftTree
	return root
}

File: tree/reverse/InvertTree.java
/*
 * @Author: yangxcc
 * @version: 1.0
 * @Date: 2022-10-24 20:49:16
 * @LastEditTime: 2023-03-03 16:24:04
 */
package tree.reverse;

import tree.TreeNode;

/**
 * leetcode 226 simple 翻转二叉树
 * 给你一棵二叉树的根节点 root ，翻转这棵二叉树，并返回其根节点。
 */
public class InvertTree {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }

        if (root.left == null && root.right == null) {
            return root;
        }
        
        TreeNode leftTmp = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(leftTmp);

        return root;
    }
}
